__TESTS__/unit/config/cloudinaryConfig.test.ts

```diff
@@ -152,4 +155,167 @@ describe('Tests for CloudinaryConfiguration', () => {
 
     expect(url).toContain('http://');
   });
-});
+  it('should allow overriding cloudName in config', () => {
+    const conf = new CloudinaryConfig({
+      cloud: {
+        cloudName: 'foo'
+      }
+    });
+
+    conf.cloud.cloudName = 'demo';
+
+    expect(conf.cloud).toEqual({
+      cloudName: 'demo'
+    });
+  });
+
+  // These tests should be "translated" to js base code when these missing features are added:
+  it.skip('should use resource_type from options', () => {
+    test_cloudinary_url('test', {
+      resource_type: 'raw'
+    }, 'https://res.cloudinary.com/test123/raw/upload/test', {});
+  });
+
+  it.skip('should support external cname with cdn_subdomain on', () => {
+    test_cloudinary_url('test', {
+      cname: 'hello.com',
+      cdn_subdomain: true
+    }, 'https://a2.hello.com/test123/image/upload/test', {});
+  });
+  it.skip('should allow to shorted image/upload urls', () => {
+    test_cloudinary_url('test', {
+      shorten: true
+    }, 'https://res.cloudinary.com/test123/iu/test', {});
+  });
+  it.skip('should support url_suffix in shared distribution', () => {
+    test_cloudinary_url('test', {
+      url_suffix: 'hello'
+    }, 'https://res.cloudinary.com/test123/images/test/hello', {});
```

Comment:
Can you please implement the full test?

You can set suffix on an asset using `image.suffix = 'foo'`, the suffix will not change anything in the URL yet (so the test will fail)